src/analysisd/testrule.c

```diff
@@ -671,3 +695,72 @@ void onexit() {
 void onsignal(__attribute__((unused)) int signum) {
     exit(EXIT_SUCCESS);
 }
+
+
+void DecodeEventchannelTestRule(Eventinfo *lf)
+{
+    cJSON *event = NULL;
+    cJSON *json_event = NULL;
+    const char *jsonErrPtr;
+    OS_XML xml;
+
+    print_out("\n**Phase 2: Completed decoding.\n");
+    print_out("       decoder: 'windows_eventchannel'");
+
+    /* Obtain event in JSON format */
+    if (!strncmp(lf->log, "{\"win\":", 7))
+    {
+        if (event = cJSON_ParseWithOpts(lf->log, &jsonErrPtr, 0), !event)
+        { print_out("\nMalformed eventchannel JSON event.\n"); }
+
+        if (json_event = cJSON_GetObjectItem(event, "win"), !json_event)
+        { print_out("\nMalformed eventchannel JSON event.\n"); }
+
+        if(printArrayEvent(lf, json_event, "system"))
+        { print_out("\nMalformed JSON: field 'system' not found.\n"); }
+
+        if(printArrayEvent(lf, json_event, "eventdata"))
+        { print_out("\nMalformed JSON: field 'eventdata' not found.\n"); }
+
+        cJSON_Delete(event);
+    }
+    /* Obtain event in XML format */
+    else if (!strncmp(lf->log, "<Event", 6))
+    {
+        if (OS_ReadXMLString(lf->log, &xml) < 0 || DecodeWinevt(lf))
+        { print_out("\nMalformed eventchannel XML event.\n"); }
+    }
+    else
+    { print_out("\nMalformed eventchannel event.\n"); }
+
+    lf->decoder_info = winevt_decoder;
+}
+
+static int printArrayEvent (Eventinfo *lf, cJSON *event, char *name)
+{
+    int result = 1;
+    int json_size, i;
+    cJSON *item = NULL;
+    cJSON *obj = NULL;
+    size_t array_size = 300;
+    char name_item[array_size];
+
+    if(obj = cJSON_GetObjectItem(event, name), !obj)
+    { return result; }
+
+    json_size = cJSON_GetArraySize(obj);
```

Comment:
The result size should be checked, and if the value equals 0 or lower, return error.